pkg/cache/clusterqueue.go

```diff
@@ -306,6 +328,16 @@ func (c *clusterQueue) inactiveReason() (string, string) {
 	return kueue.ClusterQueueActiveReasonReady, "Can admit new workloads"
 }
 
+func (c *clusterQueue) isTASViolated() bool {
+	if !features.Enabled(features.TopologyAwareScheduling) || len(c.tasFlavors) == 0 {
+		return false
+	}
+	return c.HasParent() ||
```

Comment:
Could we verify this during the validation?
I'm wondering if we can follow the similar approach with the MK managedBy validations: https://github.com/kubernetes-sigs/kueue/blob/5203ac9eb69226a1ae3b6ba6d361eaed7ef3dc8c/pkg/controller/jobs/job/job_webhook.go#L90

Reply:
Well, I guess we could but there are downsides: 
1. this mechanism would be per Job, so plenty of code which is going to be removed when we start to support the features
2. it would be inconsistent with other CQ validation scenarios when we take the lazy validation approach
3. it requires the webhooks to access the cache (yes, we did it for managedBy, but there it was last resort option because the managedBy field is immutable)

Reply:
I was wondering if we could support the validations in the CQ, RF, and Topology webhook. 
Could you expand the reason why the validations should be done in the Kueue supported Jobs webhooks?

Reply:
Oh, sorry, I was mistaken (was thinking about the annotations somehow we discussed in the morning). Indeed, it would be validation in the cq_webhook only for the scenarios in this PR.

Reply:
Still, we already have mechanics to validate CQ and deactivate, so the approach implemented here is more consistent with what we have.

Reply:
I discussed this with @mimowo offline.
Technically, we can validate these points in the CQ, RF, and Topology validating webhooks.

But, the webhook validation solutions have some issues like performance and complexity due to cross Object validations. Those fields are especially not immutable, and we can assume the validations will be performed many times. That could bring us cluster performance issues.

So, we decided to select this non-webhook solutions.